headers.h

```diff
@@ -11,15 +11,30 @@
 #define HAVE_SETSID
 
 /* OS-specific includes */
-#if defined(__GLIBC__)
+#if defined(__GLIBC__) || defined(__CYGWIN__)
+# define HAVE_UNISTD_H
+# define HAVE_SYS_IOCTL_H
 # define HAVE_SYS_SYSMACROS_H
 # define HAVE_ERROR_H
+# define HAVE_TTYNAME
 #endif
 
 /* Now we actually include crap ;) */
+#ifdef HAVE_UNISTD_H
+# include <unistd.h>
+#endif
 #ifdef HAVE_ERROR_H
 # include <error.h>
 #endif
+#ifdef HAVE_SYS_IOCTL_H
+# include <sys/ioctl.h>
+#endif
 #ifdef HAVE_SYS_SYSMACROS_H
 # include <sys/sysmacros.h>
 #endif
+
+#if defined(__GNUC_STDC_INLINE__) && (__GNUC_STDC_INLINE__ - 0)
```

Comment:
Does this work with Clang?

Reply:
@grobian AFAIR, you're using Clang: Do you know whether it does accept the `#if (__GNUC_STDC_INLINE__ - 0)` preprocessor construct?

Reply:
```
% echo "#if defined(__GNUC_STDC_INLINE__) && (__GNUC_STDC_INLINE__ - 0)\n#error yes\n#endif" | clang -E -
# 1 "<stdin>"
# 1 "<built-in>" 1
# 1 "<built-in>" 3
# 340 "<built-in>" 3
# 1 "<command line>" 1
# 1 "<built-in>" 2
# 1 "<stdin>" 2
<stdin>:2:2: error: yes
#error yes
 ^

1 error generated.
```

So I guess yes?